hackathon_site/dashboard/frontend/src/api/types.ts

```diff
@@ -0,0 +1,44 @@
+/** Generics */
+export interface APIListResponse<T> {
```

Comment:
What do you mean "over the any type"?

Tying something as `foo: any` in typescript is not very helpful, since it doesn't tell you anything.

These generics (or templates as they would be called in C++) let us write a generic interface that we can make more specific by passing in additional type annotations. So something typed as with `response: APIListResponse<Hardware>` will know that `response.results` will be a hardware array, and if you try to access say `response.results[0].in_stock`, that will cause a type error since `in_stock` isn't a field on `Hardware`.

For a better explanation, read the typescript docs on generics: https://www.typescriptlang.org/docs/handbook/2/generics.html